src/param_curve.rs

```diff
@@ -137,12 +137,27 @@ pub trait ParamCurveArea {
     fn signed_area(&self) -> f64;
 }
 
+/// The nearest position on a curve to some point.
+///
+/// This is returned by [`ParamCurveNearest::nearest`]
+#[derive(Debug, Clone, Copy)]
+pub struct Nearest {
+    /// The square of the distance from the nearest position on the curve
+    /// to the given point.
+    pub distance: f64,
```

Comment:
Because this is distance squared (to save a sqrt, as most often in hit testing scenarios a lot of these will be compared), maybe `distance_sq` or `distance2`? Calling it "distance" to me raises the risk it will be misinterpreted.